main/templates/main/partials/notebooks-table.html

```diff
@@ -0,0 +1,91 @@
+<style>
+.sort {
+  padding:8px 30px;
+  border-radius: 6px;
+  border:none;
+  display:inline-block;
+  color:#fff;
+  text-decoration: none;
+  background-color: #28a8e0;
+  height:30px;
+}
+.sort:hover {
+  text-decoration: none;
+  background-color:#1b8aba;
+}
+.sort:focus {
+  outline:none;
+}
+.sort:after {
+  display:inline-block;
+  width: 0;
+  height: 0;
+  border-left: 5px solid transparent;
+  border-right: 5px solid transparent;
+  border-bottom: 5px solid transparent;
+  content:"";
+  position: relative;
+  top:-10px;
+  right:-5px;
+}
+.sort.asc:after {
+  width: 0;
+  height: 0;
+  border-left: 5px solid transparent;
+  border-right: 5px solid transparent;
+  border-top: 5px solid #fff;
+  content:"";
+  position: relative;
+  top:4px;
+  right:-5px;
+}
+.sort.desc:after {
+  width: 0;
+  height: 0;
+  border-left: 5px solid transparent;
+  border-right: 5px solid transparent;
+  border-bottom: 5px solid #fff;
+  content:"";
+  position: relative;
+  top:-4px;
+  right:-5px;
+}
+</style>
+
+<div id="notebooks">
+  <div class="input-group-sm" style="margin-right:10px;display:inline-block;">
+    <input class="search form-control" placeholder="Search" style="border-radius:4px;"/>
+  </div>
+  {% for source in data_sources %}
+  <button class="btn btn-default btn-sm source-filter filter-{{ source|cut:" " }}">{{ source }}</button>
```

Comment:
These buttons don't have any padding, that's why they are running into each other when stacked vertically. For that reason I made the extra option`padded` in the OH CSS that gives 5px space.